src/Core/Geographic/Coordinates.js

```diff
@@ -148,8 +150,38 @@ function _convert(coordsIn, newCrs, target) {
         }
 
         if (coordsIn.crs in proj4.defs && newCrs in proj4.defs) {
-            const p = instanceProj4(coordsIn.crs, newCrs).forward([coordsIn._values[0], coordsIn._values[1]]);
-            return target.set(newCrs, p[0], p[1], coordsIn._values[2]);
+            var val0 = coordsIn._values[0];
+            var val1 = coordsIn._values[1];
+
+            // Verify that coordinates are stored in reference unit.
+            const refUnit = crsToUnit(coordsIn.crs);
+            if (coordsIn._internalStorageUnit != coordsIn.crs) {
+                if (coordsIn._internalStorageUnit == UNIT.DEGREE && refUnit == UNIT.RADIAN) {
+                    val0 = coordsIn.longitude(UNIT.RADIAN);
+                    val1 = coordsIn.latitude(UNIT.RADIAN);
+                } else if (coordsIn._internalStorageUnit == UNIT.RADIAN && refUnit == UNIT.DEGREE) {
+                    val0 = coordsIn.longitude(UNIT.DEGREE);
+                    val1 = coordsIn.latitude(UNIT.DEGREE);
+                }
+            }
+
+            // there is a bug for converting anything from and to 4978 with proj4
+            // https://github.com/proj4js/proj4js/issues/195
+            if (newCrs == 'EPSG:4978') {
+                // the workaround is to use an intermediate projection, like EPSG:4326
```

Comment:
This comment applies to both the `if` and `else if` case, so you could move it 1 line above.